packages/insured-bridge-relayer/src/index.ts

```diff
@@ -51,129 +55,155 @@ export async function run(logger: winston.Logger, l1Web3: Web3): Promise<void> {
     // TODO: Add a method to fetch all registered chainIDs from bridge admin to let the bot default to all chains when
     // the config does not include activatedChainIds.
 
-    // For now, this bot only supports 1 L2 chain. In future we need to update the bot to create n number l2Clients for
-    // each L2 client. Then, create n instances of `Relayer`.
-    assert(config.activatedChainIds.length == 1, "bot only supports running on 1 l2 at a time for now");
-
-    // Construct a web3 instance running on L2.
-    const l2Web3 = getWeb3ByChainId(config.activatedChainIds[0]);
-    const latestL2BlockNumber = await l2Web3.eth.getBlockNumber();
-    const l2StartBlock = Math.max(0, latestL2BlockNumber - config.l2BlockLookback);
-    const fallbackL2Web3s = getRetryWeb3sByChainId(config.activatedChainIds[0]);
-    // Note: This will not construct duplicate Web3 objects for URL's in the Retry config that are the same as the
-    // one used to construct l2Web3.
-    logger.debug({
-      at: "AcrossRelayer#index",
-      message: `Constructed ${fallbackL2Web3s.length} fallback L2 web3 providers`,
-    });
-    const l2Client = new InsuredBridgeL2Client(
-      logger,
-      l2Web3,
-      await l1Client.getL2DepositBoxAddress(config.activatedChainIds[0]),
-      config.activatedChainIds[0],
-      l2StartBlock,
-      null,
-      fallbackL2Web3s
-    );
-
-    // Update the L2 client and filter out tokens that are not whitelisted on the L2 from the whitelisted L1 relay list.
-    const filteredL1Whitelist = await pruneWhitelistedL1Tokens(
-      logger,
-      l1Client,
-      l2Client,
-      config.whitelistedRelayL1Tokens
+    const relayers = await Promise.all(
```

Comment:
This looks like a lot of changes -- it really just moves all of this logic into a `Promise.all(...map(` block. For vast majority of the logic, it's just an indent change.